dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserAccountService.java

```diff
@@ -67,92 +67,115 @@ public class DefaultUserAccountService implements UserAccountService {
   private static final int MAX_LENGTH = 80;
   private static final int MAX_PHONE_NO_LENGTH = 30;
 
+  @Override
+  public void validateSelfRegUser(UserRegistrationParams userRegParams, String remoteAddress)
+      throws BadRequestException, IOException {
+    log.info("Validating user info");
+    validateCaptcha(userRegParams.getRecaptchaResponse(), remoteAddress);
+    validateSelfRegUser(userRegParams);
+
+    if (!configService.getConfiguration().selfRegistrationAllowed()) {
+      throw new BadRequestException("User self registration is not allowed");
+    }
+  }
+
+  @Override
+  public void validateInvitedUser(UserRegistrationParams params, String remoteIpAddress)
+      throws BadRequestException, IOException {
+    validateCaptcha(params.getRecaptchaResponse(), remoteIpAddress);
+    validateInvitedUser(params);
+  }
+
   @Override
   @Transactional
-  public void createAndAddSelfRegisteredUser(
-      SelfRegistrationParams selfRegForm, HttpServletRequest request) {
+  public void createSelfRegisteredUser(SelfRegistrationParams params, HttpServletRequest request) {
     UserRole userRole = configService.getConfiguration().getSelfRegistrationRole();
     OrganisationUnit orgUnit = configService.getConfiguration().getSelfRegistrationOrgUnit();
 
     User user = new User();
-    user.setUsername(selfRegForm.getUsername());
-    user.setFirstName(selfRegForm.getFirstName());
-    user.setSurname(selfRegForm.getSurname());
-    user.setEmail(selfRegForm.getEmail());
-    user.setPhoneNumber(selfRegForm.getPhoneNumber());
+    user.setUsername(params.getUsername());
+    user.setFirstName(params.getFirstName());
+    user.setSurname(params.getSurname());
+    user.setEmail(params.getEmail());
+    user.setPhoneNumber(params.getPhoneNumber());
     user.getOrganisationUnits().add(orgUnit);
     user.getDataViewOrganisationUnits().add(orgUnit);
 
-    userService.encodeAndSetPassword(user, selfRegForm.getPassword());
+    userService.encodeAndSetPassword(user, params.getPassword());
     user.setSelfRegistered(true);
     user.getUserRoles().add(userRole);
 
     userService.addUser(user, new SystemUser());
     log.info("Created new user");
 
-    authenticate(user.getUsername(), selfRegForm.getPassword(), user.getAuthorities(), request);
+    authenticate(user.getUsername(), params.getPassword(), user.getAuthorities(), request);
   }
 
   @Override
-  public void validateUserFormInfo(SelfRegistrationParams selfRegForm, HttpServletRequest request)
-      throws BadRequestException, IOException {
-    log.info("Validating user info");
-    validateCaptcha(selfRegForm.getRecaptchaResponse(), request);
-    validateUserForm(selfRegForm);
+  @Transactional
+  public void updateInvitedRegisteredUser(
```

Comment:
Is this when the invited user confirms and the account is created? What if we actually call it that? `confirmInvite`?

Reply:
The User is already created by the admin is my understanding, so this is actually updating the user. I don't mind what it's called.

Reply:
updated